Source/PickUpAndHaul/JobDriver_HaulToInventory.cs

```diff
@@ -102,14 +141,11 @@ public override IEnumerable<Toil> MakeNewToils()
 				}
 			}
 		};
+	}
 
-		//maintain cell reservations on the trip back
-		//TODO: do that when we carry things
-		//I guess that means TODO: implement carrying the rest of the items in this job instead of falling back on HaulToStorageJob
-		yield return TargetB.HasThing ? Toils_Goto.GotoThing(TargetIndex.B, PathEndMode.ClosestTouch)
```

Comment:
I completely deleted this code, because it didn't make much sense to me. `TargetB` is not where anything is guaranteed to be dropped off at - the `Unload` job driver owns that responsibility - maintaining reservations seems moot, if that was a desired feature - wouldn't you do this in the Unload JobDriver?

Reply:
Mm. I think this is a more general issue:

When we use a storage building, we store a `(StoreTarget, CellAllocation)` pair. Where `CellAllocation` is presumed to be for a particular item. This means we store a `int capacity`. This `capacity` is only correct for a particular item, not *any* item. 

This means, for example - if we haul leather, to a cell with 225 capacity (3x free cells) - and we only have, i.e. 10 leather. We would then see 215 free slots. 

If we then went to haul a bunch of single-slot items, i.e. apparel. We could attempt to haul ~215 individual apparel items into this slot - when we should only be able to store 2. Because fundamentally we are storing things by cell.